server/src/test/java/org/elasticsearch/search/aggregations/metrics/TDigestStateReleasingTests.java

```diff
@@ -32,14 +36,75 @@ public TDigestStateReleasingTests(TDigestState.Type digestType) {
         this.digestType = digestType;
     }
 
+    public void testCreateOfType() {
+        testCircuitBreakerTrip(circuitBreaker -> TDigestState.createOfType(circuitBreaker, digestType, 100));
+    }
+
+    public void testCreateUsingParamsFrom() {
+        testCircuitBreakerTrip(circuitBreaker -> {
+            try (TDigestState example = TDigestState.createOfType(newLimitedBreaker(ByteSizeValue.ofMb(100)), digestType, 100)) {
+                return TDigestState.createUsingParamsFrom(example);
+            }
+        });
```

Comment:
If you use the cranky breaker service it'll blow up *randomly* and you can assert that you always freed the memory, no matter the random explosion. I rather like it for this sort of thing.


Which one of your tests hits a the breaker exception you are guarding?

Reply:
> Which one of your tests hits a the breaker exception you are guarding?

The `testReadWithData()`. The reserve is only called when it has data in it. So I added that one explicitly.

About the cranky breaker, I'm not sure we want it to break randomly. This has multiple potential breaking points I wanted to test (Add memory and release on error, on every constructor).

The cranky breaks 1/20 times. If we're fine with this one (The full suite takes around 1 second on my machine), I would leave it as-is. It tests every potential breaking point (_Divided in 16 random seeds_).
Is it too much? Not sure

Reply:
>About the cranky breaker, I'm not sure we want it to break randomly. This has multiple potential breaking points I wanted to test (Add memory and release on error, on every constructor).

That's where the cranky is really good at.

Reply:
Just updated it to use the cranky breaker. It will fail ~1/5 times, so I guess it's ok. I suppose even 1/20 times would be ok for something like this.
I guess I was reinventing the wheel here XD